addons/knowledge/models/res_users.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api
+
+
+class Users(models.Model):
+    _name = 'res.users'
+    _inherit = 'res.users'
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        users = super(Users, self).create(vals_list)
+        users.filtered(lambda user: not user.partner_share)._generate_tutorial_articles()
+        return users
+
+    def _generate_tutorial_articles(self):
+        article_template = self.sudo().env.ref('knowledge.knowledge_article_welcome', raise_if_not_found=False)
+        if not article_template:
+            return
+        articles_to_create = [{
+            'name': '%s %s' % (article_template.name, user.name),
+            'body': article_template.body.replace('Hello there', 'Hello there, %s' % user.name),
```

Comment:
This will be broken if the body is translated, not sure that is something we want to do.

Reply:
Indeed this went under the radar ... we took some time to actually remove that hack and instead generate the body based on a qweb template. Which means translations will be available, based on created user language.

Also added a context key to skip onboarding article creation, just because we know this will be necessary for some production environment. Thanks for your in-depth review !